File: backend/src/main/java/br/ufpr/sept/so2/modules/solicitacoes/api/dto/ProtocoloPublicoResponse.java
package br.ufpr.sept.so2.modules.solicitacoes.api.dto;

import br.ufpr.sept.so2.modules.solicitacoes.domain.Solicitacao;

public record ProtocoloPublicoResponse(
        String protocolo,
        String tipoNome,
        String estado,
        String hashSha256
) {

    public static ProtocoloPublicoResponse from(Solicitacao solicitacao) {
        return new ProtocoloPublicoResponse(
                solicitacao.getProtocolo().getValor(),
                solicitacao.getTipoNome(),
                solicitacao.getEstado(),
                solicitacao.getHashSha256()
        );
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/solicitacoes/api/dto/ProtocoloPublicoResponse.kt
package br.ufpr.sept.so2.modules.solicitacoes.api.dto

import br.ufpr.sept.so2.modules.solicitacoes.domain.Solicitacao

data class ProtocoloPublicoResponse(
    val protocolo: String,
    val tipoNome: String,
    val estado: String,
    val hashSha256: String?,
) {
    companion object {
        @JvmStatic
        fun from(solicitacao: Solicitacao): ProtocoloPublicoResponse =
            ProtocoloPublicoResponse(
                solicitacao.protocolo.valor,
                solicitacao.tipoNome,
                solicitacao.estado,
                solicitacao.hashSha256,
            )
    }
}
